src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.tsx

```diff
@@ -345,8 +345,20 @@ function ContactMethodDetailsPage({route}: ContactMethodDetailsPageProps) {
                         hasMagicCodeBeenSent={hasMagicCodeBeenSent}
                         handleSubmitForm={(validateCode) => validateSecondaryLogin(loginList, contactMethod, validateCode, formatPhoneNumber)}
                         validateError={!isEmptyObject(validateLoginError) ? validateLoginError : getLatestErrorField(loginData, 'validateCodeSent')}
-                        clearError={() => clearContactMethodErrors(contactMethod, !isEmptyObject(validateLoginError) ? 'validateLogin' : 'validateCodeSent')}
-                        sendValidateCode={() => requestContactMethodValidateCode(contactMethod)}
+                        clearError={() => {
+                            // When removing unverified contact methods, the ValidateCodeActionForm unmounts and triggers clearError.
+                            // This causes loginData to become an object, which makes sendValidateCode trigger, so we add this check to prevent clearing the error.
+                            if (!loginData.partnerUserID) {
```

Comment:
We should use the ref to prevent the [issue](https://github.com/Expensify/App/issues/71741) while removing the contact method from other device.